src/gateway.h

```diff
@@ -35,10 +35,20 @@ struct gateway {
 	uint64_t protocol;           /* Protocol format version */
 	uint64_t client_id;
 	struct id_state random_state; /* For generating IDs */
+	/* handle_exec_sql uses this to defer some work to the next loop
+	 * iteration. */
+	uv_timer_t sched;
 };
 
+/**
+ * Initialize a gateway.
+ *
+ * Passing NULL for the `loop` is permitted. Currently the loop is only used
+ * optionally to break potential recursion in handle_exec_sql.
```

Comment:
I think it is better if the comment does not refer to the internal function name `handle_exec_sql` but instead talks about the higher level idea.
```suggestion
 * Passing NULL for the `loop` is permitted. Currently the loop is only used
 * optionally to break potential recursion when handling a query that bundles
 *  several sql statements delimited by semicolons.
```